### PR TITLE
Add OpenTelemetry Metrics example

### PR DESCRIPTION
**Pull request by KarstenSchnitter**
### Description
Provides a small Docker Compose setup to show metrics ingestion with Data Prepper. This example is similar to the traces and logs examples. It also shows the metrics Data Prepper emits on its metrics endpoint.
 
### Issues Resolved
no issue known
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: examples/metrics-ingestion-otel/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    depends_on:
+      - opensearch
+  opensearch:
+    container_name: opensearch
+    image: docker.io/opensearchproject/opensearch:latest
+    environment:
+      - discovery.type=single-node
+      - bootstrap.memory_lock=true # along with the memlock settings below, disables swapping
+      - "OPENSEARCH_JAVA_OPTS=-Xms512m -Xmx512m" # minimum and maximum Java heap size, recommend setting both to 50% of system RAM
+      - "OPENSEARCH_INITIAL_ADMIN_PASSWORD=Developer@123"
```
Comment (dlvenable) [inline examples/metrics-ingestion-otel/docker-compose.yaml:28]: We should use the same password in all of our examples. Originally I was suggesting `yourStrongPassword123!` as the [initial blog post](https://opensearch.org/blog/replacing-default-admin-credentials/) used that.

But, I did just try typing it and it is a bit of a pain. :)

Should we update all usages to `Developer@123` instead?

File: examples/metrics-ingestion-otel/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+To generate some demo data, the OpenTelemetry Collector uses its host metrics receiver to acquire cpu and memory metrics on the machine it is running on.
+Additionally, it scrapes the Prometheus metrics endpoint of the Data Prepper instance.
+This also let's you investigate the Data Prepper metrics in OpenSearch.
+
+To run:
+
+1. Run `docker compose up`
+2. Wait for everything to come up.
+3. Log into OpenSearch Dashboards at <http://localhost:5601>.
```
Comment (dlvenable) [inline examples/metrics-ingestion-otel/README.md:17]: Let's let the community know the username and password to provide.

File: examples/metrics-ingestion-otel/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+4. Create an Index Pattern for index `otel_metrics` choosing `time` as the time field.
```
Comment (dlvenable) [inline examples/metrics-ingestion-otel/README.md:18]: Should the pattern by `otel_metrics*`?

**Review by dlvenable — CHANGES_REQUESTED** [review]
Nice, thank you @KarstenSchnitter !

File: examples/metrics-ingestion-otel/docker-compose.yaml (hunk 1 above)
Comment (KarstenSchnitter) [reply examples/metrics-ingestion-otel/docker-compose.yaml:28]: I would vote for using `Developer@123` in all examples. Some (trace-analytics) are still trying to configure `admin`, which is rejected by OpenSearch as too weak.

File: examples/metrics-ingestion-otel/README.md (hunk 3 above)
Comment (KarstenSchnitter) [reply examples/metrics-ingestion-otel/README.md:18]: I deliberately chose not to use a roll-over index. Therefore, no wildcard is rerquired. The idea was, to keep the example as simple as possible. Do you want to change it to a daily index?

File: examples/metrics-ingestion-otel/README.md (hunk 2 above)
Comment (KarstenSchnitter) [inline examples/metrics-ingestion-otel/README.md:17]:
```suggestion
3. Log into OpenSearch Dashboards at <http://localhost:5601> using username `admin` and password `Developer@123`.
```

File: examples/metrics-ingestion-otel/docker-compose.yaml (hunk 1 above)
Comment (dlvenable) [reply examples/metrics-ingestion-otel/docker-compose.yaml:28]: Sounds good!

**Pull request merged**